### PR TITLE
Bump to opentelemetry-mapping-go v0.11.0

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

- Updates opentelemetry-mapping-go libraries to v0.11.0
- Updates code according to refactor related to upcoming metric in the exporter:
    - DataDog/opentelemetry-mapping-go/pull/230
    - DataDog/opentelemetry-mapping-go/pull/231
    - DataDog/opentelemetry-mapping-go/pull/229
- Use `logs.Translator` for logs implementation

### Motivation

On the Datadog exporter, we want to report a metric related to missing hostnames in resources. For now, this will be a Datadog exporter only thing, so here the telemetry settings are set to noop, either explicitly (as in the trace agent) or implicitly (as in the metrics and logs where we disable telemetry).

### Describe how to test/QA your changes

This is an internal change only, with no user-facing impact.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/trace/config/config.go [hunk 1]
```diff
@@ -72,6 +74,9 @@ type OTLP struct {
 	// If spans have the "sampling.priority" attribute set, probabilistic sampling is skipped and the user's
 	// decision is followed.
 	ProbabilisticSampling float64
+
+	// AttributesTranslator specifies an OTLP to Datadog attributes translaator.
```
Comment (mackjmr) [inline pkg/trace/config/config.go:78]:
```suggestion
	// AttributesTranslator specifies an OTLP to Datadog attributes translator.
```

**Review by mackjmr — APPROVED** [review]

**Review by vickenty — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

**Review by ahmed-mez — APPROVED** [review]

File: .copyright-overrides.yml [hunk 2]
```diff
@@ -160,6 +160,8 @@ github.com/dgraph-io/badger:
  - "Copyright 2020 Dgraph Labs, Inc. and Contributors"
  - "Copyright 2021 Dgraph Labs, Inc. and Contributors"
 github.com/mostynb/go-grpc-compression: "Copyright 2023 Mostyn Bramley-Moore."
+github.com/pjbgf/sha1cd: "Copyright 2009 The Go Authors"
```
Comment (chouquette) [inline .copyright-overrides.yml:163]: Shouldn't this mention Paulo Gomes in addition to the Go authors?

Comment (mx-psi) [reply .copyright-overrides.yml]: Added. Looks like #21864 added this override yesterday, so I just added Paulo Gomes in that entry

**Review by chouquette — APPROVED** [review]

**Pull request merged**